File: LeetCode/55. 跳跃游戏.ts
function canJump(nums: number[]): boolean {
    return dfs(nums, 0);
};

function dfs(nums: number[], pos: number): boolean {
    if (pos + nums[pos] >= (nums.length-1)) return true;
    let flag: boolean = false;
    for (let i = 0;i < nums[pos];i ++) {
        flag = flag || dfs(nums, pos+i);
        if (flag) return true;
    }
    return false;
}

File: LeetCode/55.跳跃游戏.py
class Solution:
    cache = []
    def canJump(self, nums: List[int]) -> bool:
        for i in range(len(nums)):
            self.cache.append(0)
        return self.dfs(nums, 0)
        
    def dfs(self, nums: List[int], pos: int) -> bool:
        if pos + nums[pos] >= len(nums)-1:
            return True
        if nums[pos] == 0: return False

        for i in range(nums[pos]):
            if self.cache[pos+i+1] == 1: return True
            elif self.cache[pos+i+1] == -1: continue 
            else:
                self.cache[pos+i+1] = self.dfs(nums, pos+i+1)
                if self.cache[pos+i+1] == 1: return True
        return False